### PR TITLE
MULE-13928: CorrelationId should be propagated from flow to other flo…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…ws or systems via connector operations

### CODE REVIEW

File: src/main/java/org/mule/extensions/jms/internal/config/JmsConfig.java [hunk 1]
```diff
@@ -72,6 +74,14 @@
   @ParameterGroup(name = "Producer Config", showInDsl = true)
   private JmsProducerConfig producerConfig;
 
+  /**
+   * Whether to specify a correlationId when publishing messages. This applies both for custom correlation ids specifies at the
+   * operation level and for default correlation Ids taken from the current event
+   */
+  @Parameter
+  @Optional(defaultValue = "AUTO")
+  private OutboundCorrelationStrategy sendCorrelationId = AUTO;
```
Comment (alegmarra) [inline src/main/java/org/mule/extensions/jms/internal/config/JmsConfig.java]: why `= AUTO`?

File: src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java [hunk 2]
```diff
@@ -280,7 +284,7 @@ private void doReply(JmsResponseMessageBuilder messageBuilder,
         LOGGER.debug(format("Begin reply to destination [%s] of type [%s]", destinationName, replyToTopic ? TOPIC : QUEUE));
       }
 
-      Message message = messageBuilder.build(connection.getJmsSupport(), session.get(), config);
+      Message message = messageBuilder.build(connection.getJmsSupport(), ALWAYS, correlationInfo, session.get(), config);
```
Comment (alegmarra) [inline src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java]: shouldn't this be AUTO so it checks for isOutboundCorrelationEnabled?

File: src/main/java/org/mule/extensions/jms/internal/config/JmsConfig.java (hunk 1 above)
Comment (marianogonzalez) [reply src/main/java/org/mule/extensions/jms/internal/config/JmsConfig.java]: auto is the intended default behavior

File: src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java (hunk 2 above)
Comment (marianogonzalez) [reply src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java]: yes, good catch

File: src/main/java/org/mule/extensions/jms/internal/config/JmsConfig.java (hunk 1 above)
Comment (alegmarra) [reply src/main/java/org/mule/extensions/jms/internal/config/JmsConfig.java]: I mean: `defaultValue = "AUTO"` should be enough, `sendCorrelationId = AUTO` should not be required.

**Pull request merged**